cluster/kubernetes/sync.go

```diff
@@ -2,31 +2,308 @@ package kubernetes
+	return flux.MakeResourceID(ns, kind, name)
+}
+
+// Bytes returns a byte slice description
+func (r *kuberesource) Bytes() []byte {
+	return []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  namespace: %q
+  name: %q
+`, r.obj.GetAPIVersion(), r.obj.GetKind(), r.obj.GetNamespace(), r.obj.GetName()))
+}
+
+func (r *kuberesource) Policy() policy.Set {
+	return kresource.PolicyFromAnnotations(r.obj.GetAnnotations())
+}
+
+// GetChecksum returns the checksum recorded on the resource from
+// Kubernetes, or an empty string if it's not present.
+func (r *kuberesource) GetChecksum() string {
+	return r.obj.GetAnnotations()[checksumAnnotation]
+}
+
+// GetSyncSet returns the sync set name recorded on the the resource
+// from Kubernetes, or an empty string if it's not present.
+func (r *kuberesource) GetSyncSet() string {
+	return r.obj.GetLabels()[syncSetLabel]
+}
+
+func (c *Cluster) getResourcesBySelector(selector string) (map[string]*kuberesource, error) {
+	listOptions := meta_v1.ListOptions{}
+	if selector != "" {
+		listOptions.LabelSelector = selector
+	}
+
+	resources, err := c.client.coreClient.Discovery().ServerResources()
+	if err != nil {
+		return nil, err
+	}
+
+	result := map[string]*kuberesource{}
+
+	contains := func(a []string, x string) bool {
+		for _, n := range a {
+			if x == n {
+				return true
+			}
+		}
+		return false
+	}
+
+	for _, resource := range resources {
+		for _, apiResource := range resource.APIResources {
+			verbs := apiResource.Verbs
+			if !contains(verbs, "list") {
+				continue
+			}
+
+			namespaced := apiResource.Namespaced
+
+			// get group and version
+			var group, version string
+			groupVersion := resource.GroupVersion
+			if strings.Contains(groupVersion, "/") {
+				a := strings.SplitN(groupVersion, "/", 2)
+				group = a[0]
+				version = a[1]
+			} else {
+				group = ""
+				version = groupVersion
+			}
+
+			resourceClient := c.client.dynamicClient.Resource(schema.GroupVersionResource{
+				Group:    group,
+				Version:  version,
+				Resource: apiResource.Name,
+			})
+			data, err := resourceClient.List(listOptions)
+			if err != nil {
+				return nil, err
+			}
+
+			for i, item := range data.Items {
+				apiVersion := item.GetAPIVersion()
+				kind := item.GetKind()
+
+				itemDesc := fmt.Sprintf("%s:%s", apiVersion, kind)
+				// https://github.com/kontena/k8s-client/blob/6e9a7ba1f03c255bd6f06e8724a1c7286b22e60f/lib/k8s/stack.rb#L17-L22
+				if itemDesc == "v1:ComponentStatus" || itemDesc == "v1:Endpoints" {
+					continue
+				}
+				// TODO(michael) also exclude anything that has an ownerReference (that isn't "standard"?)
+
+				res := &kuberesource{obj: &data.Items[i], namespaced: namespaced}
+				result[res.ResourceID().String()] = res
+			}
+		}
+	}
+
+	return result, nil
+}
+
+// exportResourcesInStack collates all the resources that belong to a
+// stack, i.e., were applied by flux.
+func (c *Cluster) getResourcesInSyncSet(name string) (map[string]*kuberesource, error) {
```

Comment:
I would rename this to something more descriptive like (by it's name I thought it got the resources inside a SyncSet data structure).

`getClusterSyncSetLabeledResources` ?
`getSyncSetLabeledResources` ? (which is shorter)

Reply:
It's a method on the cluster.

Reply:
Yeah, that's true